test/test_common/environment.h

```diff
@@ -72,6 +72,25 @@ class TestEnvironment {
    */
   static std::string getCheckedEnvVar(const std::string& var);
 
+  /**
+   * Obtain an appropriate base-id (for use as the base id option for an Envoy
+   * server). Each test that needs a base id should invoke this method with
+   * a different value for base_test_id. The base test id is mutated using the
+   * following environment variables:
+   * - TEST_RANDOM_SEED to handle concurrent runs via the Bazel --runs_per_test flag
+   * - TEST_SHARD_INDEX to handle concurrent runs in coverage tests
+   *
+   * We require a unique base test id because random seeds are reused across
+   * multiple tests. For example, running:
+   *     bazel --runs_per_test=3 //test:foo_test //test:bar_test
+   * results in 3 unique seeds. Each set of tests is run once with each
+   * seed. Thus, the base test id is used to differentiate between tests run in
+   * parallel with the same seed.
+   *
+   * @param base_test_id a uint32_t, larger than 1000000, to uniquely identify a test
+   */
+  static std::string chooseBaseId(uint32_t base_test_id);
```

Comment:
It's still not clear to me what this call uses different bases in different places? Can you explain that or maybe just list here what the current allocations are?

Reply:
Switched it to just take numbers and do the 1000000 thing internally. Hopefully that makes it better. (1000000 is just to give space for a lot of random test seeds or test shards).

Reply:
Re-wrote the comments on the method as well.